mirgecom/euler.py

```diff
@@ -76,31 +75,40 @@
 class ConservedVars:  # FIXME: Name?
     r"""Resolve the canonical conserved quantities.
 
-    Get the canonical conserved quantities (mass, energy, momentum)
-    per unit volume = $(\rho,\rho{E},\rho\vec{V})$ from an agglomerated
-    object array.
+    Get the canonical conserved quantities (mass, energy, momentum,
+    and mass fractions) per unit volume = $(\rho,\rho{E},\rho\vec{V},
+    \rho{Y_s})$ from an agglomerated object array.
 
     .. attribute:: dim
 
+        Integer indicating spatial dimension of the state
+
     .. attribute:: mass
 
-        Mass per unit volume
+        :class:`~meshmode.dof_array.DOFArray` for the scalar mass per unit volume
 
     .. attribute:: energy
 
-        Energy per unit volume
+        :class:`~meshmode.dof_array.DOFArray` for total energy per unit volume
 
     .. attribute:: momentum
 
-        Momentum vector per unit volume
+        Object array of :class:`~meshmode.dof_array.DOFArray` for momentum density
+        of shape ``(ndim,)``
+
+    .. attribute:: mass_fractions
+
+        Object array of :class:`~meshmode.dof_array.DOFArray` for species mass
+        fractions, $\rho~Y_\alpha$ shape=(nspecies,).
 
     .. automethod:: join
     .. automethod:: replace
     """
 
-    mass: np.ndarray
-    energy: np.ndarray
-    momentum: np.ndarray
+    mass: DOFArray
+    energy: DOFArray
+    momentum: DOFArray
```

Comment:
```suggestion
    momentum: np.ndarray
```

(`mass_fractions` should be an `np.ndarray` too, but I'm not sure if that's a one line fix.)

Reply:
I think it is fixed up now.